pkgs/applications/misc/rofi-rbw/default.nix

```diff
@@ -1,28 +1,69 @@
-{ lib, buildPythonApplication, fetchFromGitHub, configargparse, setuptools, poetry-core, rbw }:
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, configargparse
+, setuptools
+, poetry-core
+, rbw
+
+, waylandSupport ? false
+, wl-clipboard
+, wtype
+
+, x11Support ? false
+, xclip
+, xdotool
+}:
 
 buildPythonApplication rec {
   pname = "rofi-rbw";
-  version = "1.1.0";
+  version = "1.2.0";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "fdw";
     repo = "rofi-rbw";
     rev = "refs/tags/${version}";
-    hash = "sha256-5K6tofC1bIxxNOQ0jk6NbVoaGGyQImYiUZAaAmkwiTA=";
+    hash = "sha256-6ZM+qJvVny/h5W/+7JqD/CCf9eayExvZfC/z9rHssVU=";
   };
 
   nativeBuildInputs = [
     setuptools
     poetry-core
   ];
 
+  buildInputs = [
+    rbw
+  ] ++ lib.optionals waylandSupport [
+    wl-clipboard
+    wtype
+  ] ++ lib.optionals x11Support [
+    xclip
+    xdotool
+  ];
+
   propagatedBuildInputs = [ configargparse ];
 
   pythonImportsCheck = [ "rofi_rbw" ];
 
+  wrapper_paths = [
+    rbw
+  ] ++ lib.optionals waylandSupport [
+    wl-clipboard
+    wtype
+  ] ++ lib.optionals x11Support [
+    xclip
+    xdotool
+  ];
+
+  wrapper_flags = if waylandSupport
+  then "--typer wtype --clipboarder wl-copy"
+  else if x11Support
+  then "--typer xdotool --clipboarder xclip"
+  else "";
```

Comment:
Use `lib.optionalString` instead.

P.S.: are X11 and Wayland mutually exclusive?

Reply:
They are not. those options only provide wrappers to start the right tools in the right context.